### PR TITLE
Player model

### PR DESCRIPTION
**Pull request by owenl131**
Add in player model for movement.

Outstanding tasks:
- Make bullet come out from gun area
- Make movement more fluid
- Fix spinning issues

### CODE REVIEW

**Comment by daniellimws** [comment]
- Bullet doesn't exactly come out from gun when facing north.

shooting

- Feels weird when the player doesn't follow the mouse when moving.

Movement feels better now tho.

File: Scenes/MainLevel.tscn [hunk 1]
```diff
@@ -13,7 +13,7 @@ shader = ExtResource( 8 )
 shader_param/scale = 2
 shader_param/threshold = 2.0
 shader_param/color = Color( 0, 0, 0, 1 )
-shader_param/enabled = true
+shader_param/enabled = false
```
Comment (daniellimws) [inline Scenes/MainLevel.tscn:16]: This is because of the thing you sent in telegram?

File: Scripts/RunningCharacter.cs [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+    bool stopped = false;
+
+    // Called when the node enters the scene tree for the first time.
+    public override void _Ready()
+    {
+        weaponPosition = GetNode<MeshInstance>("WeaponPosition");
+
+        player = GetNode<AnimationPlayer>("AnimationPlayer");
+        player.Play("animation_run_pistol_1");
```
Comment (daniellimws) [inline Scripts/RunningCharacter.cs:18]: Start with running animation?

File: Scripts/RunningCharacter.cs [hunk 3]
```diff
@@ -0,0 +1,64 @@
+using System;
+using Godot;
+
+public class RunningCharacter : Spatial
```
Comment (daniellimws) [inline Scripts/RunningCharacter.cs:4]: Can rename as `RunningCharacterAnimation`? Or something that indicates it's for animation?

**Review by daniellimws — CHANGES_REQUESTED** [review]

**Comment by daniellimws** [comment]
Crashes when using DashSkill on an enemy

File: Scripts/Player.cs [hunk 4]
```diff
@@ -161,6 +162,12 @@ public void Move(Vector2 weightedDirection, float delta)
         }
     }
 
+    public void WeaponFired(Projectile projectile) 
+    {
+        GD.Print(projectile.Velocity.Length());
```
Comment (daniellimws) [inline Scripts/Player.cs]: Harro

File: Scripts/Player.cs [hunk 5]
```diff
@@ -161,6 +162,12 @@ public void Move(Vector2 weightedDirection, float delta)
         }
     }
 
+    public void WeaponFired(Projectile projectile) 
+    {
+        GD.Print(projectile.Velocity.Length());
+        ApplyCentralImpulse(-10 * projectile.Mass * projectile.Velocity);
```
Comment (daniellimws) [inline Scripts/Player.cs]: What is 10?

**Review by daniellimws — CHANGES_REQUESTED** [review]

File: Scenes/MainLevel.tscn (hunk 1 above)
Comment (owenl131) [reply Scenes/MainLevel.tscn:16]: No this mainly cos when the object got more details like the hands and legs then the outline is a bit problematic.

File: Scripts/RunningCharacter.cs (hunk 3 above)
Comment (owenl131) [reply Scripts/RunningCharacter.cs:4]: Its technically the mesh + animation. Can call it RunningSprite maybe?

File: Scenes/MainLevel.tscn (hunk 1 above)
Comment (daniellimws) [reply Scenes/MainLevel.tscn:16]: oh ok

File: Scripts/RunningCharacter.cs (hunk 3 above)
Comment (owenl131) [reply Scripts/RunningCharacter.cs:4]: Or MainCharacterSprite?

**Comment by owenl131** [comment]
> Crashes when using DashSkill on an enemy

I think fixed alr

File: Scripts/Player.cs (hunk 5 above)
Comment (owenl131) [reply Scripts/Player.cs]: Some multiplier to make it look more obvious lol

File: Scripts/Player.cs (hunk 4 above)
Comment (owenl131) [reply Scripts/Player.cs]: Hmm i thought i fixed that. I think haven't pushed yet

File: Scripts/Player.cs (hunk 5 above)
Comment (daniellimws) [reply Scripts/Player.cs]: Lol ok. Can call it recoil multiplier or smth ah

File: Scripts/RunningCharacter.cs (hunk 3 above)
Comment (daniellimws) [reply Scripts/RunningCharacter.cs:4]: Isn't sprite a 2d image

Comment (daniellimws) [reply Scripts/RunningCharacter.cs:4]: or `MainCharacterModel` ah

Comment (owenl131) [reply Scripts/RunningCharacter.cs:4]: a computer graphic which may be moved on-screen and otherwise manipulated as a single entity.

Comment (owenl131) [reply Scripts/RunningCharacter.cs:4]: anything bah

File: Scripts/RunningCharacter.cs (hunk 2 above)
Comment (owenl131) [reply Scripts/RunningCharacter.cs:18]: Lol

Comment (owenl131) [reply Scripts/RunningCharacter.cs:18]: Superfluous line

File: Scripts/RunningCharacter.cs (hunk 3 above)
Comment (daniellimws) [reply Scripts/RunningCharacter.cs:4]: ok you just decide whichever better

File: Scripts/RunningCharacter.cs (hunk 2 above)
Comment (daniellimws) [reply Scripts/RunningCharacter.cs:18]: Lol ok

File: Scripts/Player.cs (hunk 5 above)
Comment (daniellimws) [reply Scripts/Player.cs]: Zis

**Comment by daniellimws** [comment]
@jeffsieu any reviews?

**Comment by jeffsieu** [comment]
ok wait ah

**Review by jeffsieu — APPROVED** [review]

**Comment by jeffsieu** [comment]
@daniellimws Can check again, I added the impulse multiplier variable.

**Review by daniellimws — APPROVED** [review]
K nice

**Pull request merged**